lib/sqlalchemy/dialects/postgresql/base.py

```diff
@@ -2307,6 +2316,29 @@ def visit_create_index(self, create, **kw):
 
         return text
 
+    def visit_unique_constraint(self, constraint, **kw):
+        if len(constraint) == 0:
+            return ""
+        nulls_not_distinct_option = ""
+        nulls_not_distinct = constraint.dialect_options["postgresql"][
+            "nulls_not_distinct"
+        ]
+        if nulls_not_distinct is True:
+            nulls_not_distinct_option = "NULLS NOT DISTINCT "
+        elif nulls_not_distinct is False:
+            nulls_not_distinct_option = "NULLS DISTINCT "
```

Comment:
**mike bayer** (zzzeek) wrote:

not that big a deal but I'd find this easier to read if there was a final:

   else:
       nulls_not_distinct_option = ""

instead of having it above

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4632